dan_layer/template_lib/src/models/confidential_proof.rs

```diff
@@ -54,14 +53,58 @@ pub struct ConfidentialStatement {
     pub encrypted_data: EncryptedData,
     #[cfg_attr(feature = "ts", ts(type = "number"))]
     pub minimum_value_promise: u64,
+    pub viewable_balance_proof: Option<ViewableBalanceProof>,
+}
+
+/// A zero-knowledge proof that a confidential resource amount is valid
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[cfg_attr(feature = "ts", derive(TS), ts(export, export_to = "../../bindings/src/types/"))]
+pub struct ViewableBalanceProof {
+    /// E = v.G + r.P
+    #[cfg_attr(feature = "ts", ts(type = "Uint8Array"))]
+    pub elgamal_encrypted: RistrettoPublicKeyBytes,
+    #[cfg_attr(feature = "ts", ts(type = "Uint8Array"))]
+    pub elgamal_public_nonce: RistrettoPublicKeyBytes,
+    #[cfg_attr(feature = "ts", ts(type = "Uint8Array"))]
+    pub c_prime: PedersonCommitmentBytes,
```

Comment:
Maybe we need a doc or something for these letters

Reply:
Cool will add on the next PR - I had copilot do them at first but they were all wrong so I removed them lol